src/middleware/authMiddleware.ts

```diff
@@ -0,0 +1,53 @@
+import { Request, Response, NextFunction } from "express";
+import jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+import userServices from "../services/usersService";
+import { UnauthorizedError } from "../error-handler/custom-errors";
+
+dotenv.config();
+
+export default function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  if (req.method === "OPTIONS") {
+    next();
+  }
+  try {
+    const accessToken = req.headers.authorization?.split(" ")[1];
+    const googleToken = req.cookies["jwt"];
```

Comment:
In our own auth we should have only our `refreshToken` and `accessToken`, google related auth should be under those fields.